app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -401,10 +401,16 @@ def traces_helpers(search:)
       end
 
       def package_names_for_job_template(action:, search:)
-        actions = ['install']
+        actions = %w(install remove update).freeze
         case action
         when 'install'
           ::Katello::Rpm.yum_installable_for_host(self).search_for(search).distinct.pluck(:name)
+        when 'remove'
+          installed_packages.search_for(search).distinct.pluck(:name)
+        when 'update'
+          pkg_names = installed_packages.search_for(search).distinct.pluck(:name)
+          upgrades = ::Katello::Rpm.installable_for_hosts([self]).select(:id, :name, :nvra, :evr).order(evr: :desc).group_by(&:name)
+          pkg_names.map { |p| upgrades[p]&.first&.nvra }
```

Comment:
When the search is blank (meaning all are selected), I'd go even further and return `[]`. That way the resulting command is just
```
yum -y update
```
instead of 
```
yum -y update package1 package2 package3 .... package572
```